### PR TITLE
check for PhysicalFileProvider

### LINKED ISSUE #47455: Json Configuration custom IFileProvider stopped working for .NET 5.0
We built support for encrypted appsettings.json in our app by using encrypted xml support built into .NET. To integrate this we use JsonConfigurationSource with a custom File Provider to return custom IFileInfo with CreateReadStream overriden to decrypt file. Like code below:

```cs
public static IConfigurationBuilder AddJsonFileProtectedByEncryptedXml(this IConfigurationBuilder builder, string path, bool optional = false, ILoggerFactory loggerFactory = null)
{
    var source = new JsonConfigurationSource
    {
        Path = path,
        // Specify file provider to use so that JsonConfigurationSource can read the decrypted form of config file.
        // Note config file does not need to be XML, only the container that contains the config file should be XML.
        FileProvider = new EncryptedXmlContainerFileProvider(loggerFactory),
        Optional = optional
    };
    builder.Add(source);
    return builder;
}
```

After upgrading to .NET 5.0 this stopped working.

Workaround: return null from PhysicalFilePath in custom IFileInfo.
https://github.com/dotnet/runtime/blob/9b9303678de0f5573fef580f3f52404ea0a20dd4/src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs#L87

### PR DESCRIPTION
**Pull request by stefannikolei**
* check that it is not a subclass of PhysicalFileProvider
fix #47455

### CODE REVIEW

**Comment by maryamariyan** [comment]
@stefannikolei thanks for the PR could you also please add a test for this?

**Review by maryamariyan — APPROVED** [review]

**Review by maryamariyan — CHANGES_REQUESTED** [review]
The PR change seems fine, @stefannikolei could you please also add a tests for this fix?

File: src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs [hunk 1]
```diff
@@ -76,9 +76,10 @@ private void Load(bool reload)
             }
             else
             {
-                static Stream OpenRead(IFileInfo fileInfo)
+                static Stream OpenRead(IFileInfo fileInfo, IFileProvider provider)
                 {
-                    if (fileInfo.PhysicalPath != null)
+                    if (fileInfo.PhysicalPath != null
+                        && provider is PhysicalFileProvider && !provider.GetType().IsSubclassOf(typeof(PhysicalFileProvider)))
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs:82]: Couldn't this just be

```suggestion
                        && provider?.GetType() == typeof(PhysicalFileProvider))
```

Also - would be great to add a comment here why we are doing this.

**Comment by safern** [comment]
@stefannikolei thanks for your PR, do you plan on addressing the feedback and adding a test case for this?

File: src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs (hunk 1 above)
Comment (davidfowl) [reply src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs:82]: Can we do this check once in the constructor

**Comment by stefannikolei** [comment]
> @stefannikolei thanks for your PR, do you plan on addressing the feedback and adding a test case for this?

Sorry missed the feedback. Gonna look at it.
Was waiting for @mcm-ham to hopefully deliver some repro code to make the test easier.

**Comment by maryamariyan** [comment]
Closing this PR. @stefannikolei this is a good change, you could create a new PR if you plan on addressing the feedback

**Pull request closed without merging**